arcgis-ios-sdk-samples/Analysis/Viewshed (GeoElement)/ViewshedGeoElementViewController.swift

```diff
@@ -0,0 +1,141 @@
+                                                       maxDistance: 250.0,
+                                                       headingOffset: 0.0,
+                                                       pitchOffset: 0.0)
+        
+        // offset viewshed observer location to top of tank
+        geoElementViewshed.offsetZ = 3.0
+        
+        // create an analysis overlay to add the viewshed to the scene view
+        let analysisOverlay = AGSAnalysisOverlay()
+        analysisOverlay.analyses.add(geoElementViewshed)
+        sceneView.analysisOverlays.add(analysisOverlay)
+
+        // set camera controller to follow tank
+        let cameraController = AGSOrbitGeoElementCameraController(targetGeoElement: tank, distance: 200.0)
+        cameraController.cameraPitchOffset = 45.0
+        sceneView.cameraController = cameraController
+        
+        sceneView.scene = scene
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // set the new waypoint
+        waypoint = mapPoint;
+        
+        // start a new timer
+        timer = Timer.scheduledTimer(timeInterval: 0.1,
+                                     target:self as Any,
+                                     selector: #selector(animate),
+                                     userInfo: nil,
+                                     repeats: true)
+    }
+    
+    func animate() {
+        guard let waypoint = waypoint,
+            let location = tank.geometry as? AGSPoint else { return }
+        
+        guard let distance = AGSGeometryEngine.geodeticDistanceBetweenPoint1(location,
+                                                                       point2: waypoint,
+                                                                       distanceUnit: AGSLinearUnit.meters(),
+                                                                       azimuthUnit: AGSAngularUnit.degrees(),
+                                                                       curveType: .geodesic) else { return }
+
+        // move toward waypoint a short distance
+        let locations = AGSGeometryEngine.geodeticMove([location],
+                                                       distance: 1.0,
+                                                       distanceUnit: AGSLinearUnit.meters(),
+                                                       azimuth: distance.azimuth1,
+                                                       azimuthUnit: AGSAngularUnit.degrees(),
+                                                       curveType: .geodesic)
+        
+        if let newLocation = locations?[0] {
+            tank.geometry = newLocation
+        }
+        
+        if let heading: Double = tank.attributes["HEADING"] as? Double {
```

Comment:
You don't need to specify the type of `heading` as it's inferred from the `as? Double`. If you prefer to have it there, that's cool, but if it were me I'd remove it.